src/main/java/htsjdk/samtools/util/TestUtil.java

```diff
@@ -89,6 +90,35 @@ public static File getTempDirecory(final String prefix, final String suffix) {
         return result;
     }
 
+    /**
+     * This method creates a temporary VCF or Bam file and its appropriately named index file, and will delete them on exit.
+     *
+     * @param prefix - The prefix string to be used in generating the file's name; must be at least three characters long
+     * @param suffix - The suffix string to be used in generating the file's name; may be null, in which case the suffix ".tmp" will be used
+     * @return A File object referencing the newly created temporary file
+     * @throws IOException - if a file could not be created.
+     */
+    public static File createTemporaryIndexedFile(final String prefix, final String suffix) throws IOException {
+        final File out = File.createTempFile(prefix, suffix);
+        out.deleteOnExit();
+        String indexFileExtension = null;
+        if (suffix.endsWith(FileExtensions.COMPRESSED_VCF)) {
+            indexFileExtension = FileExtensions.TABIX_INDEX;
+        } else if (suffix.endsWith(FileExtensions.VCF)) {
+            indexFileExtension = FileExtensions.VCF_INDEX;
+        } else if (suffix.endsWith(FileExtensions.BAM)) {
+            indexFileExtension = FileExtensions.BAM_INDEX;
+        } else if (suffix.endsWith(FileExtensions.CRAM)) {
+            indexFileExtension = FileExtensions.CRAM_INDEX;
+        }
+
+        if (indexFileExtension != null) {
+            final File indexOut = new File(out.getAbsolutePath() + indexFileExtension);
```

Comment:
Oh, I didn't think about this, but don't we need to be careful about extension vs replacement here?  Don't some of these follow replacement rules instead of additive rules.  I.e. bam.bai is ok, but it's just .crai instead of .cram.crai?

Reply:
I guess we need objects that generate index_names from file names...

Reply:
We have them, they're just not centralized.

Reply:
where?

Reply:
they are impossible to find!